src/Raven.Server/Documents/Indexes/Static/JavaScriptIndex.cs

```diff
@@ -198,13 +202,13 @@ private HashSet<CollectionName> ExecuteCodeAndCollectReferencedCollections(strin
         {
             _engine.SetValue("load", new ClrFunctionInstance(_engine, "load", LoadDocument));
 
-            _engine.Execute(Code);
+            _engine.ExecuteWithReset(Code);
 
             if (definition.AdditionalSources != null)
             {
                 foreach (var script in definition.AdditionalSources.Values)
                 {
-                    _engine.Execute(script);
+                    _engine.ExecuteWithReset(script);
```

Comment:
Should have `using (_engine.DisableMaxStatements())` here as well

Reply:
There is no limit to the number of steps it takes to _define_ maps / reduce

Reply:
`InitializeEngine` is wrapped with `DisableMaxStatements`